src/lib/encoder/Encoder.h

```diff
@@ -99,14 +100,34 @@ inline unsigned Encoder::encode(const std::string& filename, std::string output_
 	return i;
 }
 
-inline unsigned Encoder::encode_fountain(const std::string& filename, const std::function<bool(const cv::Mat&, unsigned)>& on_frame)
+inline unsigned Encoder::encode_fountain(const std::string& filename, const std::function<bool(const cv::Mat&, unsigned)>& on_frame, int compression_level)
 {
-	std::ifstream f(filename);
-	fountain_encoder_stream fes = fountain_encoder_stream::create(f, cimbar::Config::fountain_chunk_size(_eccBytes), 0); // will eventually do something clever with encode_id?
+	unsigned chunk_size = cimbar::Config::fountain_chunk_size(_eccBytes);
+
+	std::stringstream ss;
+	std::ifstream infile(filename);
+	if (compression_level <= 0)
+		ss << infile.rdbuf();
```

Comment:
Because of how wirehair works, we need to load the whole file into RAM -- this was implied within `fountain_encoder_stream::create()`, but now it is explicit.

Clearly a size limit will be a good idea.